go/arrow/avro/schema.go

```diff
@@ -0,0 +1,409 @@
+	// A time-micros logical type annotates an Avro long, where the long stores the number of microseconds
+	// after midnight, 00:00:00.000000.
+	case "time-micros":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Time64us}
+
+	// The timestamp-millis logical type represents an instant on the global timeline, independent of a
+	// particular time zone or calendar, with a precision of one millisecond. Please note that time zone
+	// information gets lost in this process. Upon reading a value back, we can only reconstruct the instant,
+	// but not the original representation. In practice, such timestamps are typically displayed to users in
+	// their local time zones, therefore they may be displayed differently depending on the execution environment.
+	// A timestamp-millis logical type annotates an Avro long, where the long stores the number of milliseconds
+	// from the unix epoch, 1 January 1970 00:00:00.000 UTC.
+	case "timestamp-millis":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_ms}
+	// The timestamp-micros logical type represents an instant on the global timeline, independent of a
+	// particular time zone or calendar, with a precision of one microsecond. Please note that time zone
+	// information gets lost in this process. Upon reading a value back, we can only reconstruct the instant,
+	// but not the original representation. In practice, such timestamps are typically displayed to users
+	// in their local time zones, therefore they may be displayed differently depending on the execution environment.
+	// A timestamp-micros logical type annotates an Avro long, where the long stores the number of microseconds
+	// from the unix epoch, 1 January 1970 00:00:00.000000 UTC.
+	case "timestamp-micros":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_us}
+	// The local-timestamp-millis logical type represents a timestamp in a local timezone, regardless of
+	// what specific time zone is considered local, with a precision of one millisecond.
+	// A local-timestamp-millis logical type annotates an Avro long, where the long stores the number of
+	// milliseconds, from 1 January 1970 00:00:00.000.
+	case "local-timestamp-millis":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_ms}
+	// The local-timestamp-micros logical type represents a timestamp in a local timezone, regardless of
+	// what specific time zone is considered local, with a precision of one microsecond.
+	// A local-timestamp-micros logical type annotates an Avro long, where the long stores the number of
+	// microseconds, from 1 January 1970 00:00:00.000000.
+	case "local-timestamp-micros":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_us}
+	//  Avro primitive type
+	default:
+		return arrow.Field{Name: node.name, Type: AvroPrimitiveToArrowType(node.ofType.(string))}
+	}
+	return arrow.Field{}
```

Comment:
shouldn't this be an error?

Reply:
the premise is that the schema would be taken from a OCF that was written with a valid schema and that there should always be a valid primitive or complex type passed into the function; I put the empty field because the compiler complained about a missing return.